src/ui/public/modules.js

```diff
@@ -1,9 +1,43 @@
+/**
+ * This module is used by Kibana to create and reuse angular modules. Angular modules
+ * can only be created once and need to have their dependencies at creation. This is
+ * hard/impossible to do in require.js since all of the dependencies for a module are
+ * loaded before it is.
+ *
```

Comment:
I don't quite follow the logic here. It sounds like Angular expects a module's dependencies to be available at creation time, and require.js can make sure that happens. What's the conflict?